text/3680-use.md

```diff
@@ -0,0 +1,463 @@
+gain one in the future.)
+
+We may want to add a `derive(Use)`, which automatically adds `Use` bounds on
+fields just as `derive(Clone)` adds `Clone` bounds on fields.
+
+## The implementation and optimization of `.use`
+
+An expression `x.use`, where `x` has type `T`, requires that `T: Use`. However,
+`x.use` does not always invoke `Clone::clone(x)`; in some cases the compiler
+can optimize away a use.
+
+If `x` is owned and statically known to be dead, the compiler will move `x`
+rather than using it. This allows functions to write `x.use` without concern
+for whether it's the last usage of `x` (e.g. when passing `x.use` to a series
+of functions). Much like a trailing comma, this allows every usage to be
+symmetric, making it easy to compare uses or add more uses afterwards. (This
+optimization also means we should generally not lint on a final use of `x.use`,
+such as we currently do with the clippy lint `redundant_clone`.)
+
+If `x` is not statically known to be dead, but *all* of the following
+conditions are met, the compiler *may* elide an `x.use` and use `&x` instead:
+- The compiler can statically see that `x` outlives the result of `x.use`,
+- The compiler can statically see that the result of `x.use` is only accessed
+  via shared reference (including methods with `&self`, dereferences via
+  `Deref`, other invocations of `.use`, `use ||` closures, or `async use`
+  blocks). Effectively, these conditions mean that the user could theoretically
+  have refactored the code to use `&x` rather than `x.use`.
+
+An example of these elisions:
+
+```rust
+fn f(obj: Arc<Object>) {
+    g(obj.use); // `g` takes `Arc<Object>`
+    h(use || obj.method()); // `Object::method` here takes `&self`
+}
+
+fn main() {
+    let obj = Arc::new(Object::new());
+    f(obj.use);
+    f(obj.use);
+    f(obj.use);
+    g(obj.use);
+}
+```
+
+If every invocation of `.use` or `use ||` here resulted in a call to
+`Clone::clone`, this program would call `Clone::clone` 10 times (and have 11
+`Arc`s to drop); however, the compiler can elide all the uses in `main` and
+have `f` work with the original Arc, resulting in only 6 calls to
+`Clone::clone` (and only 7 `Arc`s to drop). When an object is used repeatedly,
+such as in a loop, this can result in many elisions over the course of the
+program, and lower contention for atomics.
+
+If a user has an unusual `Use` type for which they wish to avoid these
+potential elisions, they can call `.clone()` directly.
+
+At any time, we could potentially instrument the compiler to detect the number
+of elided calls to `Clone::clone` in a crater run, to demonstrate the value of
+this optimization.
+
+## `use ||` closures and `async use` blocks
+
+A closure can be written as `use |args| ...`, and an async block can be written
+as `async use { ... }`, analogous to the use of `move`. (Note that `use` and
+`move` are mutually exclusive.)
+
+For any object referenced within the closure or block that a `move`
```

Comment:
I find this a little bit confusing. You can probably write this better than me but I'd be more explicit about that the closure you're referring to is  the "use" closure and I also think you wanted to say async block?. So maybe something like ...

```suggestion
For any object referenced within the use closure or use async block that a `move`
```

Still my phrasing reads weird but I guess you'd get what I meant to say :).